### PR TITLE
add some front functional

### PR DESCRIPTION
**Pull request by MiBBy-4**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -57,36 +59,84 @@ function App() {
...
-          <Route
-            path={'/login'}
-            element={<LoginPage handleLogin={handleLogin} isLoggedIn={state.isLoggedIn} />}
-          />
-          <Route
-            path={'/registration'}
-            element={<RegistrationPage handleLogin={handleLogin} isLoggedIn={state.isLoggedIn} />}
-          />
-        </Routes>
+        { state.isLoggedIn && state.customer.role === 3 ? (
```
Comment (AndreyAnischik) [inline src/App.jsx:62]: can you please name the role here and move the value to a separate file with constants?

File: src/Idea/IdeaUpdateForm.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    problem: idea.problem,
+    sphere: idea.sphere,
+    geo_focus: idea.geo_focus,
+    investor_requirements: idea.investor_requirements,
+  });
+
+  const navigate = useNavigate();
+
+  function handleIdeaChange(event) {
+    const { target: { value } } = event;
```
Comment (AndreyAnischik) [inline src/Idea/IdeaUpdateForm.jsx:20]: you can do it in one line: `const { target: { value, name } } = event;`

File: src/Idea/IdeaUpdateForm.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        <TextField id="description_input" label="description" value={state.description} variant="outlined" type="text" name="description" onChange={handleIdeaChange} />
+        <TextField id="problem_input" label="problem" value={state.problem} variant="outlined" type="text" name="problem" onChange={handleIdeaChange} />
+        <TextField id="sphere_input" label="sphere" value={state.sphere} variant="outlined" type="text" name="sphere" onChange={handleIdeaChange} />
+        <TextField id="geo_focus_input" label="geo_focus" value={state.geo_focus} variant="outlined" type="text" name="geo_focus" onChange={handleIdeaChange} />
+        <TextField id="investor_requirements_input" label="investor_requirements" value={state.investor_requirements} variant="outlined" type="text" name="investor_requirements" onChange={handleIdeaChange} />
+        <Button variant="contained" color="primary" type="submit"> Update Idea </Button>
+      </form>
+    </div>
+  );
+}
```
Comment (AndreyAnischik) [inline src/Idea/IdeaUpdateForm.jsx:53]: space

File: src/apiRequests/AdminRequests.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    idea: {
+      name: data.get('name'),
+      description: data.get('description'),
+      problem: data.get('problem'),
+      sphere: data.get('sphere'),
+      geo_focus: data.get('geo_focus'),
+      investor_requirements: data.get('investor_requirements'),
+    },
+  }, { withCredentials: true });
+}
```
Comment (AndreyAnischik) [inline src/apiRequests/AdminRequests.jsx:18]: space

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 5]
```diff
@@ -55,38 +60,85 @@ function App() {
...
-            path={'/login'}
-            element={<LoginPage handleLogin={handleLogin} isLoggedIn={state.isLoggedIn} />}
-          />
-          <Route
-            path={'/registration'}
-            element={<RegistrationPage handleLogin={handleLogin} isLoggedIn={state.isLoggedIn} />}
-          />
-        </Routes>
+        <IdeaNavbar isLoggedIn={state.isLoggedIn} customer={state.customer} />
+        { state.isLoggedIn && role === 3 ? (
```
Comment (AndreyAnischik) [inline src/App.jsx:64]: what does 3 mean?

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 5 above)
Comment (MiBBy-4) [reply src/App.jsx:64]: There are three roles: businessman, investor and admin. So, I decided to do it with int number: 1, 2, 3. Number 3 is admin

Comment (AndreyAnischik) [reply src/App.jsx:64]: it is a magic number, move it at least to constant

File: src/Idea/IdeaShow.jsx [hunk 6]
```diff
@@ -1,28 +1,64 @@
...
+
+  async function handleDeleteButton() {
+    await deleteIdea(ideaId);
+    navigate('/ideas');
+  }
+
+  function comparingDates() {
+    let date = new Date(idea.publication_period);
+    let today = new Date();
+    if (today <= date && today >= date.setDate(date.getDate() - 10)) {
```
Comment (AndreyAnischik) [inline src/Idea/IdeaShow.jsx:38]: magic number 10

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 5 above)
Comment (AndreyAnischik) [reply src/App.jsx:64]: There is still magic number here.

Comment (AndreyAnischik) [reply src/App.jsx:64]: constants should be stored in a separate file

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/apiRequests/IdeasRequests.jsx [hunk 7]
```diff
@@ -23,16 +23,21 @@ export function getIdea(ideaId) {
...
 export function changeNumberOfViews(idea) {
-  return axios.patch(`${process.env.REACT_APP_API_URL}api/v1/ideas/${idea.id}`, {
-    views: idea.views + 1,
+  const { id, views } = idea;
+  return axios.patch(`${process.env.REACT_APP_API_URL}api/v1/ideas/${id}`, {
+    views: views + 1,
   }, { withCredentials: true });
 }
 
 export function setReaction(ideaId, reaction) {
```
Comment (AndreyAnischik) [inline src/apiRequests/IdeasRequests.jsx:32]: why reactions are without `api/v1`?

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/apiRequests/IdeasRequests.jsx (hunk 7 above)
Comment (MiBBy-4) [reply src/apiRequests/IdeasRequests.jsx:32]: Cause controller of reactions was out this folder, I changed it.

File: src/Idea/IdeaNewForm.jsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    sphere: '',
+    geo_focus: '',
+    investor_requirements: '',
+    team: '',
+    next_steps: '',
+  });
+  const navigate = useNavigate();
+
+  function handleIdeaChange(event) {
+    const { target: { value } } = event;
```
Comment (AndreyAnischik) [inline src/Idea/IdeaNewForm.jsx:21]: unite value and name in one destructuring

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: src/Idea/ReactionsCount.jsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    if (reaction.liked) {
+      count += 1;
+    }
+  });
+  return count;
+}
+
+export function dislikesCount(reactions) {
+  let count = 0;
+  reactions.map((reaction) => {
```
Comment (AndreyAnischik) [inline src/Idea/ReactionsCount.jsx:15]: use reduce here

File: src/Idea/ReactionsCount.jsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
+const ONE_HUNDRED_PERCENT = 100;
+
+export function likesCount(reactions) {
+  let count = 0;
+  reactions.map((reaction) => {
```
Comment (AndreyAnischik) [inline src/Idea/ReactionsCount.jsx:5]: use reduce here

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

**Review by AndreyAnischik — APPROVED** [review]

**Pull request merged**